superset-frontend/src/views/CRUD/welcome/ActivityTableRow.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import moment from 'moment';
+import ListViewCard from 'src/components/ListViewCard';
+import { CardStyles } from '../utils';
+
+interface ActivityObjects {
```

Comment:
We should probably have separate type for each entity, then use union type for "activity object".

```
type ActivityObject = RecentSlice | RecentDashboard
```

Reply:
I agree, but again, this PR is not to improve the existing typing but to solve a bug. Refactoring and any other type of improvement is not within the scope of this PR.

Reply:
This one could be related since accurate typing may have an impact on how unique keys are chosen. Since this is not a user-facing bug, it might also make sense to do the type refactoring first (either in this or another PR) before merging this fix.

Reply:
@ktmud Do you prefer a buggy code and a more complex code than a simpler and no bug code, because the type is not perfect?

Sorry it doesn't make any sense to me.

Reply:
Please take a look at https://github.com/apache/superset/pull/13291

1. Adding typing won't necessary make the code more complex and it does help identify and simplify potential solutions (as demonstrated in my PR).
2. Current code (for React key at least) wasn't buggy to the end users.

Reply:
> 1. Adding typing won't necessary make the code more complex and it does help identify and simplify potential solutions (as demonstrated in my PR).

I didn't talk about the type but about the original component, which had 1 more component inside it. (2 components in 1).
This is a more complex component and much more difficult to test.

> 2. Current code (for React key at least) wasn't buggy to the end users.

If the user didn't see it then the bug doesn't exist?

Reply:
Hey guys sorry i haven't been following the threads. based on the conversation above, my understanding is that 
1)we are not fixing UI bug, meaning I wouldn't notice it as an end user.  2)there might be a more comprehensive solution to solve this issue. 
@yardz I prioritize code quality + cleaning up technical debt in Q1-Q2, more than even fixing an actual UI bug. Let's make sure if refractor is necessary, we are not simply putting a bandaid on the code, but solve it from the root, also not get distracted by not prioritized tasks. 
would you mind addressing @ktmud comment if that's not too much of work, or we can park this PR and move on. 

cc @rusackas

Reply:
If it does not impact the users, then there is no urgency to merge your fix. I personally feel typing is a bigger finish to catch than solving a developer warning that is not visible to users. You may disagree but there is no need to be sarcastic.

Reply:
> 1)we are not fixing UI bug, meaning I wouldn't notice it as an end user.

This is not necessarily true. Just because our test data set is not generating a noticeable error does not mean that with a different data set this error is imperceptible. This key problem generates a lot of errors when a component is rerendered.

> 2)there might be a more comprehensive solution to solve this issue.

From what I saw, there are some requests that are not correctly typed (and are in js). A definitive and responsible solution would involve much more work and several PRs. I wouldn't risk putting a "definitive" type without having done this work before it...
I believe that this component needs some refinement "cycles" to achieve the "definitive" result